### ISSUE TITLE
Unable to scan while the screen is off on Android

### ISSUE THREAD

**Issue by yasserjarouf** [xabre/xamarin-bluetooth-le#565]
With Android and while scanning, if the screen is off then the device is unable to detect or connect to devices unless the screen is lit "unlock is not necessary".

Tried to disable battery optimizations and it doesn't help. Any ideas?

**Comment by smsissuechecker** [comment]
Hi @yasserjarouf,

I'm the friendly issue checker.
It seems like (100.00 %) you haven't used our issue template :cry: I think it is very frustrating for the repository owners, if you ignore them.

If you think it's fine to make an exception, just ignore this message.
**But if you think it was a mistake to delete the template, please close the issue and create a new one.**

Thanks!

**Comment by janusw** [comment]
This simply seems to be a limitation of the Android Bluetooth stack. See e.g.:
https://stackoverflow.com/questions/48077690/ble-scan-is-not-working-when-screen-is-off-on-android-8-1-0

Note that a workaround seems to be using a ScanFilter. xamarin-bluetooth-le currently does not support this, but it was requested in #515.

**Comment by janusw** [comment]
> Note that a workaround seems to be using a ScanFilter. xamarin-bluetooth-le currently does not support this, but it was requested in #515.

Actually this is not completely true. In fact you can already use a limited scan filtering in the current version 2.1.2, by passing a service UUID to `adapter.StartScanningForDevicesAsync`. @yasserjarouf, maybe you can try that to see if it helps?

**Comment by janusw** [comment]
I have verified that the new scan-filter capabilities introduced in #578 actually make it possible to receive advertisements on Android when the screen is off. The rate at which advertisements are received is a bit lower, compared to the case where the the app is in foreground, and even lower compared to the case where the app is in the background but the screen is on (where it is already lower than in the foreground). I tried this on a Samsung Galaxy S21 with Android 12. I assume this should also work with the service-UUID filtering which is available already in 2.1.2 (but haven't tried that).

Therefore I think this issue can be closed: Scanning on Android is possible when the screen is off, one only needs to use a scan filter.
